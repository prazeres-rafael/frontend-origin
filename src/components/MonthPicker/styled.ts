import styled from 'styled-components';
import { palette } from 'utils/palette';

export const MonthParagraph = styled.p({
  color: palette.paragraph.primary,
  fontSize: 16,
  fontWeight: 600,
  textAlign: 'center',
  margin: '3px 0 2px 0',
});

export const YearParagraph = styled.p({
  color: palette.paragraph.secundary,
  fontSize: 16,
  fontWeight: 400,
  textAlign: 'center',
  margin: '2px 0 3px 0',
});

export const MonthContent = styled.div({
  display: 'flex',
  flexDirection: 'row',
  alignItems: 'center',
  border: `1px solid ${palette.border.default}`,
  borderRadius: 4,
  justifyContent: 'space-between',

  '& > div': {
    margin: '4px 0',
  },

  '& > button': {
    cursor: 'pointer',
    backgroundColor: 'transparent',
    border: 0,
  },

  '& > button > svg': {
    margin: '0 10px',
  },
});
