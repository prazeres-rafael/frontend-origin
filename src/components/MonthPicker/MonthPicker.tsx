import { Box } from '@mui/material';
import React from 'react';
import { MonthContent, MonthParagraph, YearParagraph } from './styled';
import ArrowLeft from '../../assets/svg/arrow-left.svg';
import ArrowRight from '../../assets/svg/arrow-right.svg';

export default function BasicDatePicker() {
  const months = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];

  const [monthCounter, setMonthCounter] = React.useState<number>(0);
  const [date, setDatePicker] = React.useState<Date>(new Date());

  React.useEffect(() => {
    setDatePicker(dateWithMonthsDelay(monthCounter));
  }, [monthCounter]);

  function dateWithMonthsDelay(months: number) {
    const date = new Date();
    date.setMonth(date.getMonth() + months);

    if (date < new Date()) {
      setMonthCounter(0);
      return new Date();
    } else {
      return date;
    }
  }

  const incrementMonth = () => {
    setMonthCounter(monthCounter + 1);
  };

  const decrementMonth = () => {
    setMonthCounter(monthCounter - 1);
  };

  return (
    <MonthContent>
      <button role="buttonDecrement" onClick={decrementMonth}>
        <ArrowLeft />
      </button>
      <Box>
        <MonthParagraph role="paragraphMonth">
          {months[date.getMonth()]}
        </MonthParagraph>
        <YearParagraph role="paragraphYear">{date.getFullYear()}</YearParagraph>
      </Box>
      <button role="buttonIncrement" onClick={incrementMonth}>
        <ArrowRight />
      </button>
    </MonthContent>
  );
}
