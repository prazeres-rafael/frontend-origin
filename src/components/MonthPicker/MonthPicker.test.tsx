import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';

import MonthPicker from './MonthPicker';

describe('MonthPicker', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <MonthPicker
        date={new Date()}
        decrementMonth={() => jest.fn()}
        incrementMonth={() => jest.fn()}
        month="Janeiro"
        monthByKey={() => jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
