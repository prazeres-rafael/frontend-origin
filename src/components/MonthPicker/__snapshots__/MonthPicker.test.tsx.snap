// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker should match the snapshot 1`] = `
.c1 {
  color: #1E2A32;
  font-size: 16px;
  font-weight: 600;
  text-align: center;
  margin: 3px 0 2px 0;
}

.c2 {
  color: #708797;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
  margin: 2px 0 3px 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E9EEF2;
  border-radius: 4px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > div {
  margin: 4px 0;
}

.c0 > button {
  cursor: pointer;
  background-color: transparent;
  border: 0;
}

.c0 > button > svg {
  margin: 0 10px;
}

<div>
  <div
    class="c0"
    role="monthContent"
    tabindex="0"
  >
    <button
      role="buttonDecrement"
    >
      <arrow-left.svg />
    </button>
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="c1"
        role="paragraphMonth"
      >
        Janeiro
      </p>
      <p
        class="c2"
        role="paragraphYear"
      >
        2022
      </p>
    </div>
    <button
      role="buttonIncrement"
    >
      <arrow-right.svg />
    </button>
  </div>
</div>
`;
